app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java

```diff
@@ -32,6 +32,7 @@
     PAGE_DOESNT_BELONG_TO_APPLICATION(400, 4018, "Page {0} does not belong to the application {1}",
             AppsmithErrorAction.LOG_EXTERNALLY),
     NO_DSL_FOUND_IN_PAGE(400, 4020, "The page {0} doesn't have a DSL. This is an unexpected state", AppsmithErrorAction.DEFAULT),
+    INVALID_DYNAMIC_BINDING_REFERENCE(400, 4022, "The dynamically bounded key {0} you are referring to was not found.", AppsmithErrorAction.DEFAULT),
```

Comment:
In case we are hitting this issue, there is definitely an error with the frontend DSL formation. We should log it externally so that we find out about it ASAP and fix this.
```suggestion
    INVALID_DYNAMIC_BINDING_REFERENCE(400, 4022, "The dynamically bounded key {0} you are referring to was not found.", AppsmithErrorAction.LOG_EXTERNALLY),
```